### PR TITLE
[github] Add workflow to check issues with pending validation labels

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Our Expo bot account can sometimes get rate-limited by Github. When a user creates an issue while the account is restricted, the label indicating that the report is waiting for validation (`needs validation`) is not removed and will stay there forever as we only validate the report once when the user opens the issue.

E.g. https://github.com/expo/expo/actions/runs/4410117739/jobs/7727237372

# How

Add a new workflow to check all issues with pending validation labels once a day

# Test Plan

run `act -j validate-pending-issues -s EXPO_BOT_GITHUB_TOKEN=XXX`

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: .github/workflows/check-issues-nightly.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Check issues with pending validation labels
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # 0:00 AM UTC time every day
```
Comment (Kudo) [inline .github/workflows/check-issues-nightly.yml:5]:
```suggestion
    - cron: '15 1 * * *' # 1:15 AM UTC time every day
```

0:00 would be busy time, i think we could try to adjust the time a little bit.

File: .github/workflows/check-issues-nightly.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Check issues with pending validation labels
+
+on:
```
Comment (Kudo) [inline .github/workflows/check-issues-nightly.yml:3]:
```suggestion
on:
  workflow_dispatch: {}
```

have you tried to execute this ci job? if not, maybe it's worth adding custom dispatch and trying it once.

**Review by tsapeta — APPROVED** [review]
I agree with Kudo's suggestions, looks good apart from that

File: .github/workflows/check-issues-nightly.yml (hunk 2 above)
Comment (gabrieldonadel) [reply .github/workflows/check-issues-nightly.yml:3]: Yeah I executed this locally by using [act](https://github.com/nektos/act)

File: .github/workflows/check-issues-nightly.yml (hunk 1 above)
Comment (gabrieldonadel) [reply .github/workflows/check-issues-nightly.yml]: That makes sense, I've updated it to 8:00 AM UTC (01:00 AM PST and 04:00 EST)

File: .github/workflows/check-issues-nightly.yml (hunk 2 above)
Comment (tsapeta) [reply .github/workflows/check-issues-nightly.yml:3]: It's still worth adding the `workflow_dispatch` option so we can run it with `et workflow-dispatch` command 😉

Comment (gabrieldonadel) [reply .github/workflows/check-issues-nightly.yml:3]: Sure, added

**Review by tsapeta — APPROVED** [review]

**Review by Kudo — APPROVED** [review]
thanks Gabriel!

**Pull request merged**